pytroll_collectors/tests/test_segments.py

```diff
@@ -605,6 +605,44 @@ def test_check_and_add_existing_files(self, glob, logging):
             assert "Only 'file' messages are supported." in logs
         assert logging.disable.call_count == 2
 
+    def test_messaging(self):
+        """Test that messaging is initialized correctly."""
+        with patch('pytroll_collectors.segments.create_publisher_from_dict_config') as creator:
+            with patch('pytroll_collectors.segments.ListenerContainer') as ListenerContainer:
+                self.msg_ini._setup_messaging()
+        expected_publisher = {
+            'name': 'segment_gatherer_msg',
+            'nameservers': None,
+            'port': 0,
+        }
+        creator.assert_called_with(expected_publisher)
+        creator.return_value.start.assert_called_once()
+        ListenerContainer.assert_called_once_with(
+            topics=['/foo/bar'],
+            addresses=None,
+            nameserver=None,
+            services='',
+        )
+
+    def test_messaging_disable_publisher_nameserver(self):
+        """Test that messaging is initialized correctly when nameserver connections are disabled."""
+        with patch('pytroll_collectors.segments.create_publisher_from_dict_config') as creator:
+            with patch('pytroll_collectors.segments.ListenerContainer') as ListenerContainer:
+                self.goes_ini._setup_messaging()
+        expected_publisher = {
+            'name': 'segment_gatherer_goes16',
+            'nameservers': False,
+            'port': '12345',
+        }
+        creator.assert_called_with(expected_publisher)
+        creator.return_value.start.assert_called_once()
+        ListenerContainer.assert_called_once_with(
+            topics=['/foo/bar'],
+            addresses=None,
+            nameserver=False,
+            services='',
+        )
```

Comment:
these two tests look very similar :) can they be refactored?

Reply:
Done